tdrs-backend/tdpservice/settings/common.py

```diff
@@ -5,62 +5,27 @@
 from os.path import join
 from secrets import token_urlsafe
+from typing import Any, Optional
 
 from configurations import Configuration
+from django.core.exceptions import ImproperlyConfigured
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
-# NOTE: These may be overridden by system or env_file variables
-#       Eventually we may want to consider moving these defaults into
-#       settings declarations in the class below. This will allow us to safely
-#       reference environment variables across the app as we will no longer
-#       need to rely on calling os.environ directly.
-os.environ.setdefault(
-    "ACR_VALUES",
-    "http://idmanagement.gov/ns/assurance/ial/1"
-)
-os.environ.setdefault("BASE_URL", "http://localhost:8080/v1")
-os.environ.setdefault(
-    "CLIENT_ASSERTION_TYPE",
-    "urn:ietf:params:oauth:client-assertion-type:jwt-bearer"
-)
-os.environ.setdefault(
-    "CLIENT_ID",
-    "urn:gov:gsa:openidconnect.profiles:sp:sso:hhs:tanf-proto-dev"
-)
-os.environ.setdefault("FRONTEND_BASE_URL", "http://localhost:3000")
-os.environ.setdefault(
-    "MOCK_TOKEN",
-    "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJiMmQyZDExNS0xZDdlLTQ1Nzkt"
-    "YjlkNi1mOGU4NGY0ZjU2Y2EiLCJpc3MiOiJodHRwczovL2lkcC5pbnQubG9naW4uZ292IiwiY"
-    "WNyIjoiaHR0cDovL2lkbWFuYWdlbWVudC5nb3YvbnMvYXNzdXJhbmNlL2xvYS8xIiwibm9uY2"
-    "UiOiJhYWQwYWE5NjljMTU2YjJkZmE2ODVmODg1ZmFjNzA4MyIsImF1ZCI6InVybjpnb3Y6Z3N"
-    "hOm9wZW5pZGNvbm5lY3Q6ZGV2ZWxvcG1lbnQiLCJqdGkiOiJqQzdOblU4ZE5OVjVsaXNRQm0x"
-    "anRBIiwiYXRfaGFzaCI6InRsTmJpcXIxTHIyWWNOUkdqendsSWciLCJjX2hhc2giOiJoWGpxN"
-    "2tPcnRRS196YV82dE9OeGN3IiwiZXhwIjoxNDg5Njk0MTk2LCJpYXQiOjE0ODk2OTQxOTgsIm"
-    "5iZiI6MTQ4OTY5NDE5OH0.pVbPF-2LJSG1fE9thn27PwmDlNdlc3mEm7fFxb8ZADdRvYmDMnD"
-    "PuZ3TGHl0ttK78H8NH7rBpH85LZzRNtCcWjS7QcycXHMn00Cuq_Bpbn7NRdf3ktxkBrpqyzIA"
-    "rLezVJJVXn2EeykXMvzlO-fJ7CaDUaJMqkDhKOK6caRYePBLbZJFl0Ri25bqXugguAYTyX9HA"
-    "CaxMNFtQOwmUCVVr6WYL1AMV5WmaswZtdE8POxYdhzwj777rkgSg555GoBDZy3MetapbT0csS"
-    "WqVJ13skWTXBRrOiQQ70wzHAu_3ktBDXNoLx4kG1fr1BiMEbHjKsHs14X8LCBcIMdt49hIZg"
-)
-os.environ.setdefault(
-    "OIDC_OP_AUTHORIZATION_ENDPOINT",
-    "https://idp.int.identitysandbox.gov/openid_connect/authorize"
-)
-os.environ.setdefault("OIDC_OP_ISSUER", "https://idp.int.identitysandbox.gov/")
-os.environ.setdefault(
-    "OIDC_OP_JWKS_ENDPOINT",
-    "https://idp.int.identitysandbox.gov/api/openid_connect/certs"
-)
-os.environ.setdefault(
-    "OIDC_OP_LOGOUT_ENDPOINT",
-    "https://idp.int.identitysandbox.gov/openid_connect/logout"
-)
-os.environ.setdefault(
-    "OIDC_OP_TOKEN_ENDPOINT",
-    "https://idp.int.identitysandbox.gov/api/openid_connect/token"
-)
+
+def get_required_env_var_setting(
+    env_var_name: str,
+    setting_name: Optional[str] = None
+) -> Any:
+    """Retrieve setting from environment variable, otherwise raise an error."""
+    env_var = os.getenv(env_var_name)
+    if not env_var:
+        raise ImproperlyConfigured(
```

Comment:
The previous behavior for this kind of situation was that the app would continue to startup but then when any code executed and referenced an unset environment variable an unhandled `KeyError` exception would be raised, which would lead to things like 500 errors sent to end users. With this function, any setting that uses it will cause an error to be raised on startup if it is undefined - allowing it to be caught earlier in the process with an expressive error message.

For comparison, this is what such an error could have looked like previously:
```
>>> os.environ["MY_FAKE_ENV_VAR"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.9/os.py", line 679, in __getitem__
    raise KeyError(key) from None
KeyError: 'MY_FAKE_ENV_VAR'
```
Compared to now:
```
web_1          |   File "/tdpapp/tdpservice/settings/common.py", line 31, in <module>
web_1          |     class Common(Configuration):
web_1          |   File "/tdpapp/tdpservice/settings/common.py", line 337, in Common
web_1          |     LOGIN_GOV_JWT_KEY = get_required_env_var_setting(
web_1          |   File "/tdpapp/tdpservice/settings/common.py", line 23, in get_required_env_var_setting
web_1          |     raise ImproperlyConfigured(
web_1          | django.core.exceptions.ImproperlyConfigured: Missing required setting: LOGIN_GOV_JWT_KEY - must set JWT_KEY environment variable
```